scripts/image/dither.m

```diff
@@ -0,0 +1,379 @@
+## Copyright (C) 2025 Leonardo Araujo <leolca@gmail.com>
+##
+## This program is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with this program; see the file COPYING.  If not, see
+## <https://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn  {Function File} {@var{X} = } dither (@var{RGB}, @var{map})
+## @deftypefnx {Function File} {@var{X} = } dither (@var{RGB}, @var{map}, @var{Qm}, @var{Qe})
+## @deftypefnx {Function File} {@var{BW} = } dither (@var{I})
+## Quantize an image using dithering to increase the apparent color resolution.
+##
+## @code{@var{X} = dither (@var{RGB},@var{map})} creates an indexed image
+## approximation.  It uses the color provided in the colormap, and uses dithering
+## to increase apparent color resolution.  Floyd-Steinberg error filter is:
+## @code{[   x  7]}
+## @code{[3  5  1] / 16}
+## It uses a raster scan and no weight renormalization at boundaries.
+## The default values are used: @var{Qm}=5 and @var{Qe}=8.
+##
+## @var{RGB} is a mxnx3 array with values in @code{[0, 1]} (double) or @code{[0, 255]} (uint8).
```

Comment:
Break lines in documentation before 80 (or 74) characters. (Exception are lines with `@deftypefn` or `@deftypefnx`.)